services/parser-database/parser.py

```diff
@@ -0,0 +1,230 @@
+"""Program that parses PDF documents and separates it by article for database storage.
+"""
+import logging
+import hashlib
+
+# import datetime
+# import requests
+import json
+
+
+import slate  # pylint: disable=import-error
+
+
+import retriever  # pylint: disable=import-error
+# import database  # pylint: disable=import-error
+import keywordMock  # pylint: disable=import-error
+
+
+logging.basicConfig(level=logging.INFO)
+
+
+class Document:
+    """Class for Regulations Documents"""
+
+    def __init__(self, jurisdiction, shahash, url, last_updated):
+        self.id = None
+        self.jurisdiction = jurisdiction
+        self.shahash = shahash
+        self.url = url
+        self.last_updated = last_updated
+
+    @staticmethod
+    def from_dict():
+        pass
+
+    def to_dict(self):
+        pass
+
+    def __repr__(self):
+        return f"Document(\
+                id={self.id}, \
+                jurisdiction={self.jurisdiction}, \
+                last_updated={self.last_updated}, \
+                url={self.url}, \
+                shahash={self.shahash}\
+            )"
+
+
+class Article:
+    """Class for storing articles.
+    """
+
+    def __init__(self, number, text):
+        self.number = number
+        self.text = text
+
+
+articles_in_memory = []
+keywords_in_memory = {}
+
+
+def get_article_by_number(art_num):
```

Comment:
Wdyt about using a different in-memory DS structure such as a map to store the list of articles in the following manner (I'm doing the example in `Cython++` which is a new programming language I'm working on lol):
```
unordered_map<Int, Article> articles_in_memory;

def get_article_by_number(art_num):
 if (articles_in_memory[art_num]) return articles_in_memory[art_num];
 return "No article matches such ID", 402;
```
But the equivalent of that in Python.
According to this [doc](https://wiki.python.org/moin/TimeComplexity), a `dict` DS in Python has an O(1) access time for retrieving items.

Reply:
I don´t think it is worth it to spend time on this when the DB will persist everything at a later stage...